python/ray/experimental/channel/common.py

```diff
@@ -355,8 +363,11 @@ def close(self):
 
 @DeveloperAPI
 class WriterInterface:
-    def __init__(self, output_channel: ChannelInterface):
-        self._output_channel = output_channel
+    def __init__(
+        self, output_channels: List[ChannelInterface], output_idxs: List[Optional[int]]
```

Comment:
Do we need to have multiple output channels in `WriterInterface`? Is it possible to use a single output channel where the channel itself knows which part of the output should be written to which downstream nodes?

Reply:
I suppose channel is single writer and multiple readers. In this case, how do you write different values to different readers?

Reply:
We can write different values to different readers as our chat last Friday.

We decide to provide the following semantics:

> ChannelInterface: Write the "same" data to multiple readers via one or multiple channels.

Close this comment.